File: TOH-Base/src/main/java/com/tohsoft/base/mvp/utils/language/ItemLanguageAdapter.java
package com.tohsoft.base.mvp.utils.language;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.tohsoft.base.mvp.databinding.ItemLanguageBinding;
import com.tohsoft.base.mvp.ui.BaseViewHolder;

import java.util.List;

/**
 * Created by PhongNX on 6/30/2020.
 */
public class ItemLanguageAdapter extends RecyclerView.Adapter<BaseViewHolder> {
    private List<String> mLanguageList;
    private int mSelectedIndex;

    int getSelectedIndex() {
        return mSelectedIndex;
    }

    ItemLanguageAdapter(List<String> languageList, int selectedIndex) {
        this.mLanguageList = languageList;
        this.mSelectedIndex = selectedIndex;
    }

    @NonNull
    @Override
    public BaseViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ItemLanguageBinding binding = ItemLanguageBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false);
        return new ViewHolder(binding);
    }

    public class ViewHolder extends BaseViewHolder<ItemLanguageBinding> {

        ViewHolder(ItemLanguageBinding binding) {
            super(binding);
        }

        @Override
        public void onBind(int position) {
            super.onBind(position);
            String language = mLanguageList.get(position);

            mBinding.tvItemName.setText(language);
            if (mSelectedIndex == position) {
                mBinding.radioButton.setChecked(true);
            } else {
                mBinding.radioButton.setChecked(false);
            }

            mBinding.llItem.setOnClickListener(v -> {
                mSelectedIndex = position;
                notifyDataSetChanged();
            });
            mBinding.radioButton.setOnClickListener(v -> {
                mSelectedIndex = position;
                notifyDataSetChanged();
            });
        }
    }

    @Override
    public void onBindViewHolder(@NonNull BaseViewHolder holder, int position) {
        holder.onBind(position);
    }

    @Override
    public int getItemCount() {
        return mLanguageList.size();
    }
}

File: TOH-Base/src/main/java/com/tohsoft/base/mvp/utils/language/ItemLanguageAdapter.kt
package com.tohsoft.base.mvp.utils.language

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.tohsoft.base.mvp.databinding.ItemLanguageBinding
import com.tohsoft.base.mvp.ui.BaseViewHolder

/**
 * Created by PhongNX on 6/30/2020.
 */
class ItemLanguageAdapter internal constructor(private val mLanguageList: List<String>, var selectedIndex: Int) :
        RecyclerView.Adapter<BaseViewHolder<*>>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BaseViewHolder<*> {
        val binding = ItemLanguageBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return ViewHolder(binding)
    }

    inner class ViewHolder internal constructor(binding: ItemLanguageBinding) :
            BaseViewHolder<ItemLanguageBinding>(binding) {
        override fun onBind(position: Int) {
            super.onBind(position)
            val language = mLanguageList[position]
            mBinding.tvItemName.text = language
            mBinding.radioButton.isChecked = selectedIndex == position

            mBinding.llItem.setOnClickListener { v: View? ->
                selectedIndex = position
                notifyDataSetChanged()
            }
            mBinding.radioButton.setOnClickListener { v: View? ->
                selectedIndex = position
                notifyDataSetChanged()
            }
        }
    }

    override fun onBindViewHolder(holder: BaseViewHolder<*>, position: Int) {
        holder.onBind(position)
    }

    override fun getItemCount(): Int {
        return mLanguageList.size
    }
}